sechub-scan/src/main/java/com/daimler/sechub/domain/scan/project/ScanProjectConfigAccessLevelService.java

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: MIT
+package com.daimler.sechub.domain.scan.project;
+
+import static com.daimler.sechub.sharedkernel.util.Assert.*;
+
+import java.util.Objects;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.daimler.sechub.sharedkernel.project.ProjectAccessLevel;
+import com.daimler.sechub.sharedkernel.validation.ProjectIdValidation;
+
+@Service
+public class ScanProjectConfigAccessLevelService {
+
+    public static final ProjectAccessLevel DEFAULT_ACCESS_LEVEL = ProjectAccessLevel.FULL;
+
+    private static final ScanProjectConfigID CONFIG_ID_PROJECT_ACCESS_LEVEL = ScanProjectConfigID.PROJECT_ACCESS_LEVEL;
+
+    private static final Logger LOG = LoggerFactory.getLogger(ScanProjectConfigAccessLevelService.class);
+
+    @Autowired
+    ProjectIdValidation projectIdValidation;
+
+    @Autowired
+    ScanProjectConfigService scanprojectConfigService;
+
+    public void changeProjectAccessLevel(String projectId, ProjectAccessLevel newAccessLevel, ProjectAccessLevel formerAccessLevel) {
+        /* validate */
+        notNull(newAccessLevel, "New access level may not be null!");
+        projectIdValidation.validate(projectId);
+
+        /* change configuration */
+        ProjectAccessLevel configuredAccessLevel = getProjectAccessLevelOrFallback(projectId, formerAccessLevel);
+
+        if (!Objects.equals(formerAccessLevel, configuredAccessLevel)) {
+            /*
+             * In this case this could be a race condition (two events on nearly same time
+             * happening). So we add a warn log entry.
+             */
+            LOG.warn("Configured access level was in scan domain:'{}' - but given fallback was: '{}'. This should not happen.", configuredAccessLevel,
+                    formerAccessLevel);
+        }
+
+        scanprojectConfigService.set(projectId, CONFIG_ID_PROJECT_ACCESS_LEVEL, newAccessLevel.getId());
+
+        LOG.info("Changed access level for project:{} to level:{}", projectId, newAccessLevel.getId());
+
+    }
+
+    /**
+     * Fetches project access level for given project - if no project access level
+     * is defined, default will be used, which is
```

Comment:
Where is the rest of the sentence? What is the default?

Reply:
changed to:
Fetches project access level for given project - if no project access level
is defined, default will be used, which is {@value #DEFAULT_ACCESS_LEVEL}